src/main.rs

```diff
@@ -83,25 +86,53 @@ fn first_encoder(connector_info: &ConnectorInfo) -> Option<EncoderHandle> {
     connector_info.encoders().iter().next().copied()
 }
 
+const DEFAULT_CARD_PATH: &str = "/dev/dri/card0";
+
+#[derive(Parser)]
+struct Args {
+    /// Device file representing the GPU
+    #[arg(long, default_value = DEFAULT_CARD_PATH)]
+    card_path: PathBuf,
+
+    /// Connector to use, e.g. DP-1; if not provided some connected one will be selected
+    #[arg(long)]
+    connector: Option<String>,
+}
+
 fn main() {
+    let args = Args::parse();
+
     // TODO(bschwind) - Use libdrm to iterate over available DRM devices.
-    let gpu = Card::open("/dev/dri/card0");
+    let gpu = Card::open(&args.card_path);
     dbg!(gpu.get_driver().expect("Failed to get GPU driver info"));
     dbg!(gpu.get_bus_id().expect("Failed to get GPU bus ID"));
 
     let resources = gpu.resource_handles().expect("Failed to get GPU resource handles");
 
     print_connector_info(&gpu, &resources);
 
-    let force_probe = false;
-    let Some(first_connector_handle) = first_connected_connector(&gpu) else {
-        println!("No display connected, exiting");
-        return;
-    };
+    let first_connector = {
+        let mut connectors = get_connected_connectors(&gpu);
+        let maybe_connector = if let Some(name) = &args.connector {
+            connectors.find(|info| &get_connector_name(info) == name)
+        } else {
+            connectors.next()
+        };
 
-    let first_connector = gpu
-        .get_connector(first_connector_handle, force_probe)
-        .expect("Failed to get GPU connector info");
+        let Some(first_connector) = maybe_connector else {
+            let card_path = args.card_path.to_string_lossy();
+            if let Some(name) = &args.connector {
+                println!(
+                    "Connector {name} does not exist or is not connected to {}, exiting",
+                    card_path
+                );
+            } else {
+                println!("No connector connected to {}, exiting", card_path);
+            }
+            return;
+        };
+        first_connector
```

Comment:
Jumping on a train to discuss this piece of code. :steam_locomotive: It is a nit and I'm doing mostly for the exercise (because it's already good enough).

Making `get_connected_connectors(&gpu)` return an iterator was a good step IMO.

I'd probably do this here:
```suggestion
        let card_path = args.card_path.display();
        if let Some(name) = &args.connector {
            connectors.find(|info| &get_connector_name(info) == name).expect(&format!(
                "Connector {name} does not exist or is not connected to {card_path}, exiting"
            ))
        } else {
            connectors.next().expect(&format!("No connector connected to {card_path}, exiting."))
        }
```

The `.expect(&format!("..."))` isn't great. `Result` has more handy convenience methods (and 3rd party crates), so I'd view it as a price to pay for our simplistic error handling.

Reply:
I like this, I'll take it :)

We can improve it again when we add proper error handling.